src/test/java/org/broadinstitute/hellbender/cmdline/CommandLineProgramIntegrationTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Daniel Gómez-Sánchez
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.broadinstitute.hellbender.cmdline;
+
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.GATKBaseTest;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.UUID;
+
+/**
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class CommandLineProgramIntegrationTest extends GATKBaseTest {
+
+    private static FileSystem jimfs;
+
+    @BeforeClass
+    public void setUp() {
+        jimfs = Jimfs.newFileSystem(Configuration.unix());
+    }
+
+    @AfterClass
+    public void tearDown() {
+        try {
+            jimfs.close();
+        } catch (final IOException e) {
+            log("Unable to close JimFS");
+        }
+    }
+
+    @CommandLineProgramProperties(
+            summary = "TestCreateTempPathClp",
+            oneLineSummary = "TestCreateTempPathClp",
+            programGroup = TestProgramGroup.class
+    )
+    private static final class TestCreateTempPathClp extends CommandLineProgram {
+
+        @Override
+        public Path doWork() {
+            return createTempPath(UUID.randomUUID().toString(), ".txt");
+        }
+    }
+
+    @DataProvider
+    public Object[][] tmpDirs() throws Exception {
+        return new Object[][]{
+                {createTempDir("local").toPath()},
+                {Files.createDirectory(jimfs.getPath("tmp"))}
```

Comment:
There should be 3 test cases here rather than 2 in order to cover all the branches in your code: a local directory without a `file://` scheme specified, a local directory with `file://` specified, and a Jimfs URI.

Reply:
Actually, the one with the `file://` prefix is already tested - in the test, the argument is converted with `Path.toUri().toString()`, which adds the `file://`. I refactored the method to has the `String` argument directly, and added a test case for a local file without prefix.